relbench/data/task_base.py

```diff
@@ -116,9 +120,19 @@ def test_table(self) -> Table:
                     "insufficient aggregation time."
                 )
 
+            # end_timestamp is optionally set in dataset definition
+            if self.dataset.end_timestamp is None:
+                end_timestamp = self.dataset._full_db.max_timestamp - self.timedelta
+            else:
+                end_timestamp = self.dataset.end_timestamp
+
             full_table = self.make_table(
                 self.dataset._full_db,
-                pd.Series([self.dataset.test_timestamp]),
+                pd.date_range(
+                    self.dataset.test_timestamp,
+                    end_timestamp,
+                    freq=self.timedelta,
+                ),
```

Comment:
I find this `end_timestamp` thing a bit confusing. My questions are
- Why not add this logic to `validation` as well?
- Instead of `end_timestamp`, I would add a flag like `max_eval_time_frames` (the number of unique time frames to evaluate during validation and test time) and set it to `1` for `amazon` and `stackex` and set it to some large number for `f1`.

Reply:
For validation I am just using the entire time interval `[val_timestamp, test_timestamp]`. If i add a validation `max_eval_time_frames` then there is a possibility of validation/test time leakage if this is set too big. I think it is safer (and uses the entire time interval) to always use `[val_timestamp, test_timestamp]` for validation.